src/api_manager/service_control/proto.cc

```diff
@@ -1311,32 +1311,40 @@ Status Proto::ConvertCheckResponse(const CheckResponse& check_response,
         check_response_info->service_is_activated = false;
       return Status(Code::PERMISSION_DENIED,
                     std::string("API ") + service_name +
-                        " is not enabled for the project.",
+                        " is not enabled for the consumer project.",
                     Status::SERVICE_CONTROL);
     case CheckError::RESOURCE_EXHAUSTED:
       return Status(Code::PERMISSION_DENIED, "Quota check failed.",
                     Status::SERVICE_CONTROL);
     case CheckError::PERMISSION_DENIED:
-      return Status(Code::PERMISSION_DENIED, "Permission denied.",
-                    Status::SERVICE_CONTROL);
+      return Status(
+          Code::PERMISSION_DENIED,
+          std::string("Consumer project doesn't have access to the API ") +
+              service_name,
+          Status::SERVICE_CONTROL);
     case CheckError::IP_ADDRESS_BLOCKED:
-      return Status(Code::PERMISSION_DENIED, "IP address blocked.",
+      return Status(Code::PERMISSION_DENIED,
+                    "IP address is blocked by API key.",
```

Comment:
nit: "for API key"

Reply:
it should be "by".  I changed it to "by the API key with restrictions"